app/src/main/java/com/lagradost/cloudstream3/utils/VideoDownloadManager.kt

```diff
@@ -1733,7 +1738,37 @@ object VideoDownloadManager {
         }
     }
 
-    fun deleteFileAndUpdateSettings(context: Context, id: Int): Boolean {
+    fun deleteFilesAndUpdateSettings(
+        context: Context,
+        ids: List<Int>,
+        scope: CoroutineScope,
+        onComplete: (List<Int>) -> Unit = {}
+    ) {
+        scope.launchSafe(Dispatchers.IO) {
+            val deleteJobs = ids.map { id ->
+                async {
+                    id to deleteFileAndUpdateSettings(context, id)
+                }
+            }
+            val results = deleteJobs.awaitAll()
+
+            val successfulDeletes = results.filter { it.second }.map { it.first }
+            val failedDeletes = results.filterNot { it.second }
+
+            if (failedDeletes.isNotEmpty()) {
+                failedDeletes.forEach { (id, _) ->
+                    // TODO show a toast if some failed?
```

Comment:
Yes, a toast would be good.

Reply:
> Yes, a toast would be good.

@fire-light42 

I think maybe not yet, as there seems to be another problem where for example if you have keys that exist but the files were deleted manually, it still tries to when you delete a series (which is all keys) which means it will mark as failed even when it really isn't, and thus a toast in UI may be confusing. I believe this is an unrelated issue though, but we should probably patch it at some point so non file's keys get removed...